### PR TITLE
add generic sort view

### PR DESCRIPTION
**Pull request by 2e2a**
Extends the ordering functionality of the Django `generic.ListView`:

- Applies sort from the request parameter
- Adds supported orderings dict list
- Provides function get the ordering name

### CODE REVIEW

**Comment by 2e2a** [comment]
will fix test coverage.

**Comment by 2e2a** [comment]
> will fix test coverage.

Would need to setup integration tests for this. I suggest to merge this and postpone the tests.

File: adhocracy4/contrib/views.py [hunk 1]
```diff
@@ -18,3 +19,20 @@ def raise_exception(self):
...
         """
         return self.request.user.is_authenticated()
+
+
+class SortableListView(generic.ListView):
+    """
+    ordering_current: List of possible ordering tuples (ordering,
+        ordering name).
+    """
+    orderings_supported = []
```
Comment (xi) [inline adhocracy4/contrib/views.py]: Why is this not a dict in the first place?

File: adhocracy4/contrib/views.py [hunk 2]
```diff
@@ -18,3 +19,20 @@ def raise_exception(self):
         limited by the current phase.
         """
         return self.request.user.is_authenticated()
+
+
+class SortableListView(generic.ListView):
+    """
+    ordering_current: List of possible ordering tuples (ordering,
+        ordering name).
+    """
```
Comment (xi) [inline adhocracy4/contrib/views.py]: I don't understand how this docstring is related to the code. `ordering_current` does not exist, or does it?

File: adhocracy4/contrib/views.py [hunk 3]
```diff
@@ -18,3 +19,20 @@ def raise_exception(self):
...
+    orderings_supported = []
+
+    def get_ordering(self):
+        ordering = self.request.GET.get('ordering')
+        if ordering and ordering in dict(self.orderings_supported):
+            self.ordering = [ordering]
+        return self.ordering
```
Comment (xi) [inline adhocracy4/contrib/views.py]: Getter functions should not have side effects. I think we should either set `self.ordering` in the beginning of the request (`__init__`?) or not set it at all and always use `get_ordering()`.

File: adhocracy4/contrib/apps.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class OrganisationsConfig(AppConfig):
```
Comment (xi) [inline adhocracy4/contrib/apps.py]: Should be `ContribConfig`

File: adhocracy4/contrib/views.py (hunk 1 above)
Comment (2e2a) [inline adhocracy4/contrib/views.py]: The list is used in the sort tamplate to iterate over available sortings.

**Comment by 2e2a** [comment]
reflected comments and added some tests

File: adhocracy4/contrib/views.py [hunk 5]
```diff
@@ -22,17 +21,22 @@ def raise_exception(self):
 
 
 class SortableListView(generic.ListView):
+    """Add sorting via request parameter.
+
+    Fields:
+        ordering: List containing the signle default ordering string
```
Comment (xi) [inline adhocracy4/contrib/views.py]: typo: single

**Comment by xi** [comment]
@MagdaN This just adds new functionality, so it should not break anything in euth_wagtail. However, you might want to consider using this for euth_wagtail.

**Pull request merged**